From: Bernhard Walle <bernhard@bwalle.de>
Date: Sat, 21 Jan 2012 23:40:16 +0100
Subject: [PATCH] Darwin: Compile with -fnested-functions

glibc/locale/programs/ld-ctype.c:2948: error: nested functions are disabled, use -fnested-functions to re-enable

Signed-off-by: Bernhard Walle <bernhard@bwalle.de>
---
 configure.ac |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/configure.ac b/configure.ac
index 85d3ca2..6f8d4b3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -24,6 +24,7 @@ AC_CONFIG_HEADERS([config.h])
 AC_EXEEXT
 AC_PROG_CC
 AC_PROG_LN_S
+AC_CANONICAL_HOST
 AC_C_BIGENDIAN([AC_DEFINE([BUILD_BYTE_ORDER], [__BIG_ENDIAN],
 		          [Define to your build machine's byte order])],
 	       [AC_DEFINE([BUILD_BYTE_ORDER], [__LITTLE_ENDIAN])])
@@ -59,6 +60,13 @@ LOCAL_CHECK_VAR([program_invocation_short_name], [#include <errno.h>])
 AC_CHECK_LIB([intl], [gettext])
 AC_CHECK_LIB([posix4], [nanosleep])
 
+# use -fnested-functions on Mac OS
+case ${host_os} in
+darwin*)
+    CFLAGS="${CFLAGS} -fnested-functions"
+    ;;
+esac
+
 # These two macros are taken from GCC's config/acx.m4.
 dnl Support the --with-pkgversion configure option.
 dnl ACX_PKGVERSION(default-pkgversion)
